features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/utils/SdkCoreExt.kt

```diff
@@ -44,20 +45,28 @@ internal class WriteOperation(
     fun submit() {
         sdkCore.getFeature(Feature.RUM_FEATURE_NAME)
             ?.withWriteContext { datadogContext, eventBatchWriter ->
-                try {
-                    val event = eventSource(datadogContext)
+                if (rumDataWriter is NoOpDataWriter) {
+                    sdkCore.internalLogger.log(
+                        level = InternalLogger.Level.INFO,
+                        target = InternalLogger.Target.USER,
+                        messageBuilder = { WRITE_OPERATION_IGNORED }
+                    )
```

Comment:
suggestion: we probably should also call `onSuccess/onError` here (don't know which one makes more sense), because at least for the `RumViewScope` until `sum(pending) == 0` it won't be marked as complete and thus garbage collected. We may have the case when there is `NoOpDataWriter` here and RUM pipeline working if session is not tracked, I believe.